### ISSUE TITLE
Method for converting strings to byte arrays (inverse of to_s)

### ISSUE THREAD

**Issue by dgelessus** [kaitai-io/kaitai_struct#526]
There is already a `to_s(encoding)` method to decode a byte array to a string. It would be useful to have a method for the opposite direction as well (encoding a string to a byte array).

My use case for this is writing literal byte arrays that only contain text. For example, if I have a byte array field that can contain any data (not necessarily valid ASCII) and want to check if it's equal to the ASCII string `"abc"`, I have to write `field == [0x61, 0x62, 0x63]`. It would be much more convenient and readable if I could write something like `field == "abc".to_bytes("ASCII")` instead. (I can't safely use `field.to_s("ASCII") == "abc"` - that would fail if `field` contains non-ASCII bytes.)

**Comment by GreyCat** [comment]
I heavily suspect that for your particular use case something like pythonc `b"abc"` would be the simpler and better choice — i.e. a special string literal syntax that is more comfortable to use than byte integer arrays.

**Comment by dgelessus** [comment]
Right, a byte array literal syntax like that would be helpful for writing byte arrays with many (or only) ASCII characters in them, which is indeed a common case. But I think there's still a use for encoding a string with an explicit encoding, for example when dealing with UTF-16 (`"abc".to_bytes("UTF-16-LE")` is nicer than `b"a\x00b\x00c\x00"`).
